lib/improver/tests/utilities/test_statistical_operations.py

```diff
@@ -0,0 +1,388 @@
+        percentile is passed into the plugin."""
+        percentiles_cube = set_up_percentiles_cube()
+        percentiles_cube = percentiles_cube[0]
+        msg = "Percentile coordinate has only one value. Interpolation"
+        with self.assertRaisesRegexp(ValueError, msg):
+            ProbabilitiesFromPercentiles2D(percentiles_cube)
+
+
+class Test__repr__(IrisTest):
+    """ Test string representation """
+    def test_basic(self):
+        """ Compare __repr__ string with expectation """
+        new_name = "probability"
+        test_cube = set_up_percentiles_cube()
+        inverse_ordering = False
+        expected = ('<ProbabilitiesFromPercentiles2D: percentiles_'
+                    'cube: {}, output_name: {}, inverse_ordering: {}'.format(
+                        test_cube, new_name, inverse_ordering))
+        result = str(ProbabilitiesFromPercentiles2D(test_cube,
+                                                    new_name))
+        self.assertEqual(result, expected)
+
+
+class Test_create_probability_cube(IrisTest):
+    """
+    Test creation of new probability cube based on percentile and orography
+    input cubes
+    """
+    def setUp(self):
+        """ Set up a probability cube from percentiles and orography """
+        self.percentiles_cube = set_up_percentiles_cube()
+        self.percentile_coordinate = find_percentile_coordinate(
+            self.percentiles_cube)
+        self.new_name = "probability"
+        self.pfp_instance = ProbabilitiesFromPercentiles2D(
+            self.percentiles_cube, self.new_name)
+        self.orography_cube = set_up_threshold_cube()
+
+    def test_attributes(self):
+        """ Test name and units are correctly set """
+        result = self.pfp_instance.create_probability_cube(
+            self.percentiles_cube)
+        self.assertEqual(result.units, "1")
+        self.assertEqual(result.name(), self.new_name)
+        self.assertEqual(result.attributes['relative_to_threshold'], 'below')
+
+    def test_attributes_inverse_ordering(self):
+        """Test relative_to_threshold attribute is suitable for the
+        inverse_ordering case, when it should be 'above'."""
+        self.percentiles_cube.data = np.flipud(self.percentiles_cube.data)
+        pfp_instance = ProbabilitiesFromPercentiles2D(self.percentiles_cube,
+                                                      self.new_name)
+        result = pfp_instance.create_probability_cube(self.percentiles_cube)
+        self.assertEqual(result.attributes['relative_to_threshold'], 'above')
+
+    def test_coordinate_collapse(self):
+        """ Test probability cube has no percentile coordinate """
+        result = self.pfp_instance.create_probability_cube(
+            self.percentiles_cube)
+        with self.assertRaises(CoordinateNotFoundError):
+            find_percentile_coordinate(result)
+
+
+class Test_percentile_interpolation(IrisTest):
```

Comment:
These tests look good. It might be nice to have one more test showing the result if every single percentile value was the same as an edge case (which is very possible at the moment).